x/bridge/types/genesis.go

```diff
@@ -15,11 +15,12 @@ import (
 // later stages, before running the network.
 func DefaultGenesis() *GenesisState {
 	return &GenesisState{
-		Params:                  DefaultParams(),
-		AssetsLockedSequenceTip: sdkmath.NewInt(0),
-		SourceBtcToken:          "",
-		Erc20TokensMappings:     nil,
-		InitialBtcSupply:        sdkmath.NewInt(0),
+		Params:                    DefaultParams(),
+		AssetsLockedSequenceTip:   sdkmath.NewInt(0),
+		AssetsUnlockedSequenceTip: sdkmath.NewInt(-1),
```

Comment:
I think we should set the tip to `0` initially.
I know this will start numbering `AssetsUnlocked` events from `1` and not `0` (the rip is incremented before being used to assign sequence number for the event), but that's how it works in the `MezoBridge` contract:
- we declare [sequence](https://github.com/thesis/mezo-portal/blob/f2c30f6649c6e63dbec48ab97958a17c95691eea/solidity/contracts/MezoBridge.sol#L18) and its initialized with `0`.
- we [pre-increment the sequence](https://github.com/thesis/mezo-portal/blob/f2c30f6649c6e63dbec48ab97958a17c95691eea/solidity/contracts/MezoBridge.sol#L65) before assigning its value to the event.

This has an advantage: the current value of the tip tells us how many `AssetsUnlock` we have processed. Initially `0`, then `1`, `2`, etc.

Reply:
@jeremyletang 
Could we also have a check for a negative value during genesis, similar to [check of the assets unlocked tip](https://github.com/mezo-org/mezod/blob/b8419881a9a4ae7165dae5912a4ac69f1d3f8555/x/bridge/types/genesis.go#L35)

Reply:
sure thing.